imxrt1060-hal/src/gpio.rs

```diff
@@ -160,12 +167,26 @@ where
         // The input / output direction. When switching across fast / normal, keep the
         // same direction.
         copy_bits!(GDIR, self.mask());
+
+        // Interrupt configuration is preserved when switching fast / normal.
+        if <P as Pin>::Offset::USIZE < 16 {
+            copy_bits!(ICR1, self.mask());
+        } else {
+            copy_bits!(ICR2, self.mask());
+        }
```

Comment:
Is this necessary?!

GPIO1-5 are the ones that can transition to fast (GPIO6-9). Are they maybe all in `ICR1` so we can skip the offset check?

Reply:
Selecting either ICR1 or ICR2, based on the pin number, is correct. The ICR field for `GPIO1_23` is in ICR2, at bit 15. If we become a fast GPIO (`GPIO6_23`), the ICR field is still in ICR2, at bit 15. It's just in a different register block.